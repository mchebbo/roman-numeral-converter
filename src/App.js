import React, { useState } from "react";
import RomanHelper from "./romanHelper";
import "./App.css";

function App() {
  const [arabicValue, setArabicValue] = useState(0);
  const [romanValue, setRomanValue] = useState("");

  const numeralIsValid = num => num >= 1 && num <= 3999;

  const handleArabicChange = event => {
    const value = event.target.value;
    setArabicValue(value);

    if (numeralIsValid(value)) {
      const romanValue = RomanHelper.toRoman(value);
      setRomanValue(romanValue);
    } else {
      setRomanValue("");
    }
  };

  const handleRomanChange = event => {
    const value = event.target.value;

    setRomanValue(value);
  };
  return (
    <div className="App">
      <h1 className="App-title">Modern Roman Numeral Converter</h1>
      <div className="App-container">
        <h2>Arabic style</h2>
        <input
          type="number"
          value={arabicValue}
          onChange={handleArabicChange}
        />
        {!numeralIsValid(arabicValue) && (
          <p className="text-error">The number must be between 1 and 3999</p>
        )}
      </div>
      <div className="App-container">
        <h2>Roman Style</h2>
        <input value={romanValue.toUpperCase()} onChange={handleRomanChange} />
      </div>
    </div>
  );
}

export default App;
