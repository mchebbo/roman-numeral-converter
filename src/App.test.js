import React from "react";
import App from './App'
import { create } from "react-test-renderer";

describe("App component", () => {
  test("Matches the snapshot", () => {
    const component = create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});