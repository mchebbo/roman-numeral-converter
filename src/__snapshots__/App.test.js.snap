// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="App-title"
  >
    Modern Roman Numeral Converter
  </h1>
  <div
    className="App-container"
  >
    <h2>
      Arabic style
    </h2>
    <input
      onChange={[Function]}
      type="number"
      value={0}
    />
    <p
      className="text-error"
    >
      The number must be between 1 and 3999
    </p>
  </div>
  <div
    className="App-container"
  >
    <h2>
      Roman Style
    </h2>
    <input
      onChange={[Function]}
      value=""
    />
  </div>
</div>
`;
